src/js/jsx/sections/libraries/assets/ColorTheme.jsx

```diff
@@ -29,56 +29,57 @@ define(function (require, exports, module) {
         FluxMixin = Fluxxor.FluxMixin(React),
         classnames = require("classnames"),
         tinycolor = require("tinycolor");
-        
-    var AssetButtons = require("jsx!./AssetButtons");
+
+    var ColorModel = require("js/models/color");
+
+    var AssetSection = require("jsx!./AssetSection");
 
     var ColorTheme = React.createClass({
         mixins: [FluxMixin],
 
-        _handleAdd: function () {
-            // Do something with color here
-        },
-        
         /**
-         * Handle select element event. 
-         * 
+         * Apply color to the selected layers.
+         *
          * @private
          */
-        _handleSelect: function () {
-            if (this.props.onSelect) {
-                this.props.onSelect(this.props.element);
-            }
+        _handleApply: function (colorData) {
+            var color = new ColorModel({ r: colorData.value.r, g: colorData.value.g, b: colorData.value.b });
+
+            this.getFlux().actions.libraries.applyColor(color);
         },
 
         render: function () {
             var element = this.props.element,
                 colorSwatches = element.getPrimaryRepresentation().getValue("colortheme", "data").swatches;
-                
-            var colorSwatchComponents = colorSwatches.map(function (color) {
-                var colorHex = tinycolor(color[0].value).toHexString().toUpperCase();
-                return (<div key={colorHex} style={{ background: colorHex }} title={colorHex}/>);
+
+            var colorSwatchComponents = colorSwatches.map(function (colors) {
+                var colorData = colors[0],
+                    colorHex = tinycolor(colorData.value).toHexString().toUpperCase();
+
+                return (<div key={colorHex}
```

Comment:
If the color theme has the same colors, this is going to have identical keys and React will be mad at us. Maybe use the `index` coming from the map function here instead?